browser-client.cpp

```diff
@@ -198,6 +198,30 @@ bool BrowserClient::OnProcessMessageReceived(
 	case ControlLevel::Basic:
 		if (name == "saveReplayBuffer") {
 			obs_frontend_replay_buffer_save();
+		} else if (name == "setBrowserSize") {
+			int width = input_args->GetInt(1);
+			int height = input_args->GetInt(2);
+			if (!width)
+				width = (int)round(input_args->GetDouble(1));
+			if (!height)
+				height = (int)round(input_args->GetDouble(2));
+			if (width > 0 && height > 0) {
+				obs_data_t *s =
+					obs_source_get_settings(bs->source);
+				if (s &&
+				    (obs_data_get_int(s, "width") != width ||
+				     obs_data_get_int(s, "height") != height)) {
+					obs_data_set_int(s, "width", width);
+					obs_data_set_int(s, "height", height);
+					obs_source_update(bs->source, nullptr);
```

Comment:
I think this isn't really how `obs_source_update` is meant to be used.
Sure, you *can* do it like this, but really you're meant to pass a new `obs_data_t` object containing the new settings (in place of the `nullptr`), instead of modifying the object obtained by `obs_source_get_settings` directly.

And while you're at it, you should probably use the C++-RAII-types (`OBSData` / `OBSDataAutoRelease`) in C++-code.